rules/static.go

```diff
@@ -826,6 +844,64 @@ func (r *DefaultRuleRenderer) StaticManglePreroutingChain(ipVersion uint8) *Chai
 	}
 }
 
+func (r *DefaultRuleRenderer) StaticManglePostroutingChain(ipVersion uint8) *Chain {
+	rules := []Rule{}
+
+	// Accept immediately if IptablesMarkAccept is set.  Our filter-FORWARD chain sets this for
+	// any packets that reach the end of that chain.  The principle is that we don't want to
+	// apply normal host endpoint policy to forwarded traffic.
+	rules = append(rules, r.acceptAlreadyAccepted()...)
+
+	// Similarly, avoid applying normal host endpoint policy to IPVS-forwarded traffic.
+	// IPVS-forwarded traffic is identified by having a non-zero endpoint ID in the
+	// IptablesMarkEndpoint bits.  Note: we only need this check for when net.ipv4.vs.conntrack
+	// is enabled.  When net.ipv4.vs.conntrack is disabled (which is the default),
+	// IPVS-forwarded traffic will fail the ConntrackState("DNAT") match below, and so would
+	// avoid normal host endpoint policy anyway.  But it doesn't hurt to have this additional
+	// check even when not strictly needed.
+	if r.KubeIPVSSupportEnabled {
+		rules = append(rules,
+			Rule{
+				Match:  Match().MarkNotClear(r.IptablesMarkEndpoint),
+				Action: r.filterAllowAction,
+			},
+		)
+	}
+
+	// At this point we know that the packet is not forwarded, so it must be originated by a
+	// host-based process or host-networked pod.
+
+	if ipVersion == 4 {
+		r.addIPIPVXLANEgressAllowRules(&rules)
+	}
+
```

Comment:
filterOutput has rules here to catch traffic going to a local workload.  I thought we didn't need that here, but in fact I now suspect we do, so that that traffic doesn't hit policy for the "all-interfaces" host endpoint.  That also means another FV case: Host -> service IP (iptables DNAT) -> local workload with "all-interfaces" host endpoint.

Reply:
Turns out we already have FV for this case, and it was already passing because cali-to-host-endpoint internally includes a check against applying host endpoint policy to traffic going to a local workload.  So we don't need those rules here.